engine/access/rpc/backend/script_executor_test.go

```diff
@@ -0,0 +1,287 @@
+
+func (s *ScriptExecutorSuite) SetupTest() {
+	s.log = unittest.Logger()
+	s.chain = flow.Emulator.Chain()
+
+	s.reporter = syncmock.NewIndexReporter(s.T())
+	s.indexReporter = index.NewReporter()
+	err := s.indexReporter.Initialize(s.reporter)
+	require.NoError(s.T(), err)
+
+	blockchain := unittest.BlockchainFixture(10)
+	s.headers = newBlockHeadersStorage(blockchain)
+	s.height = blockchain[0].Header.Height
+	s.reporter.On("LowestIndexedHeight").Return(s.height, nil)
+
+	entropyProvider := testutil.EntropyProviderFixture(nil)
+	entropyBlock := mock.NewEntropyProviderPerBlock(s.T())
+	entropyBlock.
+		On("AtBlockID", testifyMock.AnythingOfType("flow.Identifier")).
+		Return(entropyProvider).
+		Maybe()
+
+	s.snapshot = snapshot.NewSnapshotTree(nil)
+	s.vm = fvm.NewVirtualMachine()
+	s.vmCtx = fvm.NewContext(
+		fvm.WithChain(s.chain),
+		fvm.WithAuthorizationChecksEnabled(false),
+		fvm.WithSequenceNumberCheckAndIncrementEnabled(false),
+	)
+
+	s.dbDir = unittest.TempDir(s.T())
+	db := pebbleStorage.NewBootstrappedRegistersWithPathForTest(s.T(), s.dbDir, s.height, s.height)
+	pebbleRegisters, err := pebbleStorage.NewRegisters(db)
+	s.Require().NoError(err)
+	s.registerIndex = pebbleRegisters
+
+	derivedChainData, err := derived.NewDerivedChainData(derived.DefaultDerivedDataCacheSize)
+	s.Require().NoError(err)
+
+	indexerCore, err := indexer.New(
+		s.log,
+		metrics.NewNoopCollector(),
+		nil,
+		s.registerIndex,
+		s.headers,
+		nil,
+		nil,
+		nil,
+		nil,
+		s.chain,
+		derivedChainData,
+		nil,
+	)
+	s.Require().NoError(err)
+
+	s.scripts = execution.NewScripts(
+		s.log,
+		metrics.NewNoopCollector(),
+		s.chain.ChainID(),
+		entropyBlock,
+		s.headers,
+		indexerCore.RegisterValue,
+		query.NewDefaultConfig(),
+		derivedChainData,
+		true,
+	)
+	s.bootstrap()
+}
+
+func (s *ScriptExecutorSuite) TestExecuteAtBlockHeight() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	script := []byte("access(all) fun main() { }")
+	var scriptArgs [][]byte
+	var expectedResult = []byte("{\"type\":\"Void\"}\n")
+
+	s.Run("test scrypt execution without version control", func() {
+		scriptExec := NewScriptExecutor(s.log, nil, uint64(0), math.MaxUint64)
+		s.reporter.On("HighestIndexedHeight").Return(s.height+1, nil)
+
+		err := scriptExec.Initialize(s.indexReporter, s.scripts)
+		s.Require().NoError(err)
+
+		res, err := scriptExec.ExecuteAtBlockHeight(ctx, script, scriptArgs, s.height)
+		s.Assert().NoError(err)
+		s.Assert().NotNil(res)
+		s.Assert().Equal(expectedResult, res)
+	})
+
+	s.Run("test scrypt execution with version control with compatible version", func() {
+		// Set up a mock version beacons storage.
+		versionBeacons := storageMock.NewVersionBeacons(s.T())
+		// Mock the Highest method to return a version beacon with a specific version.
+		versionBeacons.
+			On("Highest", testifyMock.AnythingOfType("uint64")).
+			Return(func(height uint64) (*flow.SealedVersionBeacon, error) {
+				if height == s.height {
```

Comment:
similar comment here and elsewhere about reducing boiler plate
https://github.com/onflow/flow-go/pull/5984/files#r1645269077
https://github.com/onflow/flow-go/pull/5984/files#r1645271186